src/D2L.Security.OAuth2/Keys/Default/Data/StandardJwksProvider.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System;
+using System.Net.Http;
+using System.Threading.Tasks;
+using D2L.Security.OAuth2.Validation.Exceptions;
+using D2L.Services;
+
+namespace D2L.Security.OAuth2.Keys.Default.Data {
+	class StandardJwksProvider : IJwksProvider {
+		private readonly HttpClient m_httpClient;
+		private readonly Uri m_authEndpoint;
+
+		public StandardJwksProvider(
+			HttpClient httpClient,
+			Uri authEndpoint
+		) {
+			m_httpClient = httpClient;
+			m_authEndpoint = authEndpoint;
+		}
+
+		string IJwksProvider.Namespace => m_authEndpoint.AbsoluteUri;
+
+		async Task<JsonWebKeySet> IJwksProvider.RequestJwkAsync( string keyId ) {
+			try {

```

Comment:
This `try/catch`ery seems unnecessary.

Reply:
Copied from original JwksProvider, but I can review if it's necessary.

Reply:
The d2l provider is doing it's own http calls, this one is just calling through to the full-set method which already catches these things.